knative/client
# Add ApiServer source update and describe commands

## Proposed Changes
- Add `kn source apiserver update` and `kn source apiserver describe`
- Dont return created ApiServer source object but only error
- Align creating ApiServer source client, removes unit tests
- Add ApiServer source update command
- Rename TestMockKnClient to TestMockKnCronJobSourceClient
- Add mock client for ApiServer Source and its tests
- Add mock unit tests for create, delete and update
- Add e2e tests for ApiServer source update
  - Add a test for ApiServer source sink update
  - Verify the updated sink name after the ApiServer source is updated
  - Update resource names in existing tests
- Use builder pattern to generate ApiServer source

/lint
